lib/pos_mempool.go

```diff
@@ -277,6 +277,94 @@ func (mp *PosMempool) IsRunning() bool {
 	return mp.status == PosMempoolStatusRunning
 }
 
+// OnBlockConnected is an event handler provided by the PoS mempool to handle the blockchain
+// event where a block is connected to the tip of the blockchain. The mempool updates its
+// internal state based on the new block that has been connected.
+//
+// Whenever a block is connected, this event handler removes the block's transactions from
+// the mempool and updates the internal fee estimation to include new block.
+func (mp *PosMempool) OnBlockConnected(block *MsgDeSoBlock) {
+	mp.Lock()
+	defer mp.Unlock()
+
+	if block.Header == nil || !mp.IsRunning() {
+		return
+	}
+
+	// Remove all transactions in the block from the mempool.
+	for _, txn := range block.Txns {
+		txnHash := txn.Hash()
+
+		// This should never happen. We perform a nil check on the txn hash to avoid a panic.
+		if txnHash == nil {
+			continue
+		}
+
+		// Get the transaction from the register. If the txn doesn't exist in the register,
+		// then there's nothing left to do.
+		txn := mp.txnRegister.GetTransaction(txnHash)
```

Comment:
Ooooh - we use `mp.txnRegister.GetTransaction` instead of `mp.GetTransaction` here since `mp.GetTransaction` acquires the lock on the mempool (which would lead to a deadlock) right?

Reply:
Correct. The usage of `mp.txnRegister.GetTransaction()` here matches how we do txn removals elsewhere in this file